src/ring_theory/finiteness.lean

```diff
@@ -332,6 +332,16 @@ begin
   exact equiv (mv_polynomial R (fin (n + m))) (mv_polynomial.rename_equiv R sum_fin_sum_equiv).symm
 end
 
+/-- If `A` is an `R`-algebra and `S` is an `A`-algebra, both finitely presented, then `S` is
+  finitely presented as `R`-algebra. -/
+lemma trans [algebra A B] [is_scalar_tower R A B] (hfpA : finite_presentation R A)
+  (hfpB : finite_presentation A B) : finite_presentation R B :=
+begin
+  obtain ⟨n, I, e, hfg⟩ := iff.1 hfpB,
+  letI : is_scalar_tower R A I.quotient := is_scalar_tower.comap,
```

Comment:
Does it work without this line?

Reply:
Ah nevermind, perhaps
```suggestion
  -- note that this unfolds `algebra.comap` from the last argument of `is_scalar_tower`
  letI : is_scalar_tower R A I.quotient := is_scalar_tower.comap,
```

Reply:
Although if you added the instance I referenced in the previous PR / zulip, then you wouldn't need this.

Reply:
I tried to add
```
instance quotient.is_scalar_tower' [algebra S A] [algebra S R] [is_scalar_tower S R A]
  {I : ideal A} : is_scalar_tower S R (ideal.quotient I) :=
is_scalar_tower.of_algebra_map_eq' $ by
  rw [quotient.alg_map_eq R, quotient.alg_map_eq S, ring_hom.comp_assoc,
    is_scalar_tower.algebra_map_eq S R A]
```
to `ring_theory/ideal/operations.lean`, but then I get a `deterministic timeout` in `mv_polynomial_of_finite_presentation`, at the line `letI : ...`. I guess the Lean is now confused that there are two many `is_scalar_tower` instances...

BTW adding the comment seems a good temporary solution to me, the long term solution would be to automatize this kind of stuff.

Reply:
Would you mind opening a new PR that adds that instance, just so we have a record of what the failure looks like?

Reply:
#6717